plugins/spark/src/main/java/org/radargun/service/AbstractSparkLifecycle.java

```diff
@@ -39,6 +42,48 @@ protected void startInternal() {
          }
          throw new IllegalStateException("Failed to attach reader to log file");
       });
+      super.startInternal();
+   }
+   
+   @Override
+   public String getProcessId(Process process)  {
+      // TODO make property
+      int timeout = 5000;
```

Comment:
You should rather check current time in each loop than just sleep 5ms ten times.